discovery-provider/src/queries/queries.py

```diff
@@ -487,6 +503,19 @@ def get_feed():
         (limit, _) = get_pagination_vars()
         feed_results = sorted_feed[0:limit]
 
+        if "with_users" in request.args and request.args.get("with_users") != 'false':
+            user_id_list = list(set(map(lambda result: int(result["playlist_owner_id"]) if 'playlist_id' in result else int(result["owner_id"]), feed_results)))
```

Comment:
so i understand what is happening here but can we make this multiple lines, and add some comments explaining why we have structured things this way? 

am also curious whether there is some way to generate this list as we create the feed instead of re-traversing the feed contents at the end of everything. @SidSethi might have a bit more insight into this

Reply:
yeah we probably could do it earlier / avoid re-computing but it's so cheap to do this compared to all the queries, this will prob add a couple ms prob vs 1000s of ms already for the actual feed compute so i don't think it's important to change